src/plugins/wb-data-scrub/wb-data-scrub.js

```diff
@@ -0,0 +1,207 @@
+	attrBlocked = "data-wb-blocked",
+	attrScrubForm = "data-wbscrub-form",
+	piiModalID = componentName + "-modal",
+	i18n, i18nText,
+
+	init = function( event ) {
+		var elm = wb.init( event, componentName, selector );
+
+		if ( elm ) {
+			var	form = elm.form;
+
+			// Initialize i18n strings
+			if ( !i18nText ) {
+				i18n = wb.i18n;
+				i18nText = {
+					header: i18n( "pii-header" ),
+					intro: i18n( "pii-intro" ),
+					viewMore: i18n( "pii-view-more" ),
+					viewMoreIntro: i18n( "pii-view-more-intro" ),
+					viewMoreitem1: i18n( "pii-view-more-item-1" ),
+					viewMoreitem2: i18n( "pii-view-more-item-2" ),
+					viewMoreitem3: i18n( "pii-view-more-item-3" ),
+					viewMoreitem4: i18n( "pii-view-more-item-4" ),
+					viewMoreitem5: i18n( "pii-view-more-item-5" ),
+					yesBtn: i18n( "pii-yes-btn" ),
+					cancelBtn: i18n( "pii-cancel-btn" )
+				};
+			}
+
+			// Block form submission for Postback forms by default
+			form.setAttribute( attrBlocked, "true" );
+
+			// Ensure the form has an ID
+			form.id = form.id ? form.id : wb.getId();
+
+			if ( !form.isPIIinited ) {
+				form.addEventListener( "submit", function( event ) {
+					event.preventDefault();
+
+					// Go through form values
+					checkFormValues( form );
+
+					// Open modal
+					if ( form.PIIFields.length > 0 ) {
+						generateModal( form );
+
+						$( "#" + piiModalID ).trigger( "open.wb-lbx", [
+							[ {
+								src: "#" + piiModalID,
+								type: "inline"
+							} ],
+							true
+						] );
+					} else {
+
+						// Unblock form submission if no PII is found
+						form.removeAttribute( attrBlocked );
+
+						// Submit the form if it isn't a Postback form else trigger the postback submit
+						if ( !form.classList.contains( "wb-postback" ) ) {
+							event.currentTarget.submit();
+						} else {
+							$( form ).trigger( "wb-postback.submit", { event } );
+						}
+					}
+				} );
+
+				// This is to prevent this event listener to be added for
+				// every field with the attribute "data-rule-wbscrub"
+				form.isPIIinited = true;
+			}
+
+			wb.ready( $( elm ), componentName );
+		}
+	},
+
+	/*
+	* Log all PII positive fields inside the form's "PIIFields" property
+	* @param form: a reference to the form containing PII fields
+	*/
+	checkFormValues = function( form ) {
+		let fieldsToScrub = form.querySelectorAll( selector );
+
+		form.PIIFields = [];
+
+		// identify form elements that were assigned to be scrubbed
+		fieldsToScrub.forEach( ( field ) => {
+
+			// If the field contains PII add field to list
+			if ( wb.findPotentialPII( field.value, false ) ) {
+				let currFieldValue = field.value,
+					scrubbedFieldValue = wb.findPotentialPII( field.value, true, { replaceWith: "********" } ),
+					fieldLabel = form.querySelector( "[for=" + field.id + "] > span.field-name" ),
+					fieldLabelText = fieldLabel ? fieldLabel.innerText : form.querySelector( "[for=" + field.id + "]" ).innerText;
+
+				form.PIIFields.push( {
+					id: field.id,
```

Comment:
Just pass the DOM element reference. That way it avoid additional content finding manipulation.

```suggestion
					elm: field,
```